test/unit/serialization/compact/CompactGenericRecordTest.js

```diff
@@ -568,6 +589,62 @@ describe('CompactGenericRecordTest', function () {
             });
         });
 
+        describe('array restrictions', function () {
+            it('should not throw error if object types are equal on ARRAY_OF_COMPACT', function () {
+                (async () => {
+                    const {serializationService, schemaService} = createSerializationService(
+                        [new ArrayOfCompactSerializer(), new SampleObject1Serializer()]
+                    );
+                    const object1 = new SampleObject1('type1', Long.fromNumber(102310312));
+                    const object2 = new SampleObject1('type2', Long.fromNumber(102310312));
+                    const arrayOfObjects = [
+                        object1,
+                        object2
+                    ];
+                    const arrayOfCompactObject = new ArrayOfCompact(arrayOfObjects);
+                    await serialize(serializationService, schemaService, arrayOfCompactObject);
+                }).should.not.throw();
```

Comment:
If the test throws it will fail anyway. Therefore I don't think `(async () => {}).should.not.throw()` is necessary here.

Reply:
I removed these lines.